File: src/lambdas/normalize-input/telemetry.js
import {
  CloudWatchClient,
  PutMetricDataCommand,
} from "@aws-sdk/client-cloudwatch";

const cloudwatch = new CloudWatchClient({ apiVersion: "2010-08-01" });

export default async function send(event) {
  // This is to avoid sending metrics during the test suite
  if (event.StateMachine) {
    await cloudwatch.send(
      new PutMetricDataCommand({
        Namespace: "PRX/Porter",
        MetricData: [
          {
            MetricName: "JobsStarted",
            Dimensions: [
              {
                Name: "StateMachineArn",
                Value: event.StateMachine.Id,
              },
            ],
            Value: 1,
            Unit: "Count",
          },
          {
            MetricName: "TasksRequested",
            Dimensions: [
              {
                Name: "StateMachineArn",
                Value: event.StateMachine.Id,
              },
            ],
            Value: event.Input.Job.Tasks.length,
            Unit: "Count",
          },
        ],
      }),
    );
  }
}

File: src/containers/hls/telemetry.rb
require "aws-sdk-cloudwatch"

CLOUDWATCH = Aws::CloudWatch::Client.new

def send_start_metric
  # Count the tasks in CloudWatch Metrics
  puts JSON.dump({msg: "Sending telemetry, execution counter"})
  CLOUDWATCH.put_metric_data({
    namespace: "PRX/Porter",
    metric_data: [
      {
        metric_name: "HlsExecutions",
        dimensions: [
          {
            name: "StateMachineName",
            value: ENV["STATE_MACHINE_NAME"]
          }
        ],
        value: 1,
        unit: "Count"
      }
    ]
  })
end

def send_end_metric(duration)
  # Record HLS duration in CloudWatch Metrics
  puts JSON.dump({msg: "Sending telemetry, HLS duration", duration: "#{duration} seconds"})
  CLOUDWATCH.put_metric_data({
    namespace: "PRX/Porter",
    metric_data: [
      {
        metric_name: "HlsDuration",
        dimensions: [
          {
            name: "StateMachineName",
            value: ENV["STATE_MACHINE_NAME"]
          }
        ],
        value: duration,
        unit: "Seconds"
      }
    ]
  })
end
